### PR TITLE
Add lua support

### PR DESCRIPTION
**Pull request by JafarAbdi**
- Add lua.scm from nvim-treesitter
- Finish lua support
- Revert "Add lua.scm from nvim-treesitter" for some reason it's causing the program to panic

I'll take a look on why adding the highlights is causing the program to panic

### CODE REVIEW

File: src/tree_sitter_parser.rs [hunk 1]
```diff
@@ -325,6 +326,18 @@ pub fn from_language(language: guess::Language) -> TreeSitterConfig {
                 .unwrap(),
             }
         }
+        Lua => {
+            let language = unsafe { tree_sitter_lua() };
+            TreeSitterConfig {
+                name: "Lua",
+                language,
+                atom_nodes: (vec!["string"]).into_iter().collect(),
+                delimiter_tokens: (vec![("(", ")"), ("{", "}")])
```
[the remark this answers is not in the document]
Comment (Wilfred) [reply src/tree_sitter_parser.rs:335]: This look sensible, but for Lua you'll want `[` `]` too.

**Review by Wilfred — CHANGES_REQUESTED** [review]
Thanks! Lua is definitely a language I'd like to see in difftastic.

> I'll take a look on why adding the highlights is causing the program to panic

Sometimes the panic message is useful, so I'd look at that first. You don't have to have highlights, an empty string is fine, but of course it's nicer to have highlighting.

I think neovim may have extended the query syntax with lua scripting:

https://github.com/nvim-treesitter/nvim-treesitter/blob/3b974b7257516e4ea89d10dd9842b1035a540d7e/queries/lua/highlights.scm#L137

It might work if you remove the `#lua-match?` queries.

Otherwise, this is almost ready to merge! You can fix the formatting by running `cargo fmt`. You'll also need to update the sample files output test:

```
$ sample_files/compare_all.sh
$ cp sample_files/compare.result sample_files/compare.expected
```

This isn't super well documented at the moment, I'll look to fix that.

**Comment by JafarAbdi** [comment]
@Wilfred Are you going to squash merge this? or should I squash the last few commits?

**Comment by Wilfred** [comment]
@JafarAbdi please squash the last few commits, and then I will merge this :)

**Comment by Wilfred** [comment]
Thanks :)

**Pull request merged**